src/pages/en/guides/migration-guide/from-gatsby.md

```diff
@@ -0,0 +1,1184 @@
+---
+
+<PageHeader>
+	<a href="/">CompanyCo</a>
+	<li slot="navItems"><a href="/blog">Blog</a></li>
+</PageHeader>
+
+<!-- ... -->
+```
+See more about [specific `<slot />` usage in Astro](/en/core-concepts/astro-components/#slots) for details.
+
+### Gatsby Component Props to Astro
+
+To access specific attributes passed to your component (e.g. `<Layout title="About Me"/>`), use `Astro.props`.
+
+The following Gatsby React component:
+
+```jsx title="layout.jsx"
+const Layout = ({title}) => {
+    // ...
+}
+```
+
+Is turned into the following Astro component:
+
+```astro title="Layout.astro"
+---
+const {title} = Astro.props;
+---
+
+<!-- ... -->
+```
+
+#### Type-checking Props
+
+If you're coming from a TypeScript codebase, your Gatsby component might have an interface associated with the component's props:
+
+```jsx title="layout.jsx"
+interface LayoutProps {
+    title: string;
+}
+
+const Layout = ({title}: LayoutProps) => {
+    // ...
+}
+```
+
+In Astro, reuse the TypeScript interface and type cast the `Astro.props` destructure to enforce type safety:
+
+```astro title="Layout.astro"
+---
+interface LayoutProps {
+    title: string;
+}
+
+const {title} = Astro.props as LayoutProps;
+---
+
+<!-- ... -->
+```
+
+### Gatsby Styling to Astro
+
+Styling in both Gatsby and Astro comes in a few different flavors:
+
+- Inline styles
+- Component-specific styling
+- Global styling
+
+#### Inline Styles
+
+Convert any inline style objects in React (`style={{fontWeight: "bold", }}`) to inline HTML style attributes (`style="font-weight:bold;"`).
+
+```jsx
+// JSX
+<p style={{fontWeight: "bold"}}>Hello, world</p>
+```
+
+```astro
+// Astro
+<p style="font-weight: bold;">Hello, world</p>
+```
+
+#### Component-specific styling
+
+One of Astro's unique capabilities enables you to use standard `<style>` tags for scoped CSS styles.
+
+```astro
+<p>Hello, world</p>
+
+<!-- The following will only apply to this component -->
+<style>
+p {
+   font-weight: bold;
+}
+</style>
+```
+
+However, you can extract your styling to a [CSS Module file](https://docs.astro.build/en/guides/imports/#css-modules) and import that instead.
+
+```astro title="src/components/Hello.astro"
+---
+import styles from './Hello.module.css';
+---
+
+<p>Hello, world</p>
+```
+
+ ```css title="src/components/Hello.module.css"
+ p {
+    font-weight: bold;
+ }
+ ```
+
+#### Global Styling
+
+Global styling is applied similarly between Gatsby and Astro by importing a `css` file within a layout file.
+
+```astro title="src/layouts/Layout.astro" {2}
+---
+import "global.css";
+---
+
+<html>
+  <head>
+    <meta charset="utf-8" />
+	<link rel="icon" type="image/svg+xml" href="/favicon.svg" />
+	<meta name="viewport" content="width=device-width" />
+	<meta name="generator" content={Astro.generator} />
+	<title>Astro</title>
+  </head>
+  <body>
+    <slot/>
+  </body>
+</html>
+```
+
+```css title="src/layouts/global.css"
+p {
+    font-weight: bold;
+}
+```
+
+#### CSS Preprocessors
```

Comment:
Just calling out that this section doesn't feel very "Here's what to do to replicate your Gatsby setup in Astro."

Unless the example below is wildly different from how this is done in Gatsby, I'd consider just a sentence or two saying you CAN use them and so e.g. "You can use your existing setup" or a very quick explanation of what do DO when converting a site.